src/e2ee/matrixKeyProvider.ts

```diff
@@ -55,19 +55,24 @@ export class MatrixKeyProvider extends BaseKeyProvider {
     }
   }
 
-  private onEncryptionKeyChanged = async (
+  private onEncryptionKeyChanged = (
     encryptionKey: Uint8Array,
     encryptionKeyIndex: number,
     participantId: string,
-  ): Promise<void> => {
-    this.onSetEncryptionKey(
-      await createKeyMaterialFromBuffer(encryptionKey),
-      participantId,
-      encryptionKeyIndex,
-    );
+  ): void => {
+    createKeyMaterialFromBuffer(encryptionKey)
+      .then((keyMaterial) => {
+        this.onSetEncryptionKey(keyMaterial, participantId, encryptionKeyIndex);
 
-    logger.debug(
-      `Sent new key to livekit room=${this.rtcSession?.room.roomId} participantId=${participantId} encryptionKeyIndex=${encryptionKeyIndex}`,
-    );
+        logger.debug(
+          `Sent new key to livekit room=${this.rtcSession?.room.roomId} participantId=${participantId} encryptionKeyIndex=${encryptionKeyIndex}`,
+        );
+      })
+      .catch((e) => {
```

Comment:
Since the `then` callback doesn't return a Promise, I think this can be
```suggestion
      },
      (e) => {
```